src/tribol/geom/GeomUtilities.cpp

```diff
@@ -1207,12 +1225,13 @@ bool SegmentIntersection2D( const real xA1, const real yA1, const real xB1, cons
 } // end SegmentIntersection2D()
 
 //------------------------------------------------------------------------------
-FaceGeomError CheckPolySegs( const real* const RESTRICT x, const real* const RESTRICT y, 
-                             const int numPoints, const real tol, 
-                             real* RESTRICT * RESTRICT xnew, real* RESTRICT * RESTRICT ynew, 
-                             int& numNewPoints )
+TRIBOL_HOST_DEVICE FaceGeomError CheckPolySegs( const RealT* const x, const RealT* const y, 
+                                                const int numPoints, const RealT tol, 
+                                                RealT* xnew, RealT* ynew, 
+                                                int& numNewPoints )
 {
-   real newIDs[ numPoints ];
+   constexpr int max_nodes_per_overlap = 8;
+   RealT newIDs[ max_nodes_per_overlap ];
```

Comment:
In all locations where you set an array to some maximum length, can you verify/clarify whether we initialize the entries of those arrays? Here (below) we set the `newIDs`, but only up to `numPoints`. In principle, we should never index into that array beyond `numPoints`, but I'm wondering if it makes sense as a best practice to initialize all entries in the even that `numPoints < max_nodes_per_overlap`? Thoughts?

Reply:
Let's leave it as is for now.

More generally, we could use some container type which automatically initializes all entries.  If it's on the stack, it'll be a fixed size, but if we use dynamic memory, we can variably size it.  We'll have to look more at the performance implications of each option on each platform.

Reply:
That is fine for now.